src/textual/message_pump.py

```diff
@@ -522,10 +522,12 @@ async def dispatch_key(self, event: events.Key) -> None:
         Args:
             event (events.Key): A key event.
         """
-        key_method = getattr(self, f"key_{event.key_name}", None)
+        key_method = getattr(self, f"key_{event.key_name}", None) or getattr(
+            self, f"_key_{event.key_name}", None
+        )
         if key_method is not None:
-            if await invoke(key_method, event):
-                event.prevent_default()
+            await invoke(key_method, event)
```

Comment:
Sight change in semantics here. If there is a `key_` method defined for a given key then it won't attempt to try the base classes.